src/main/java/com/teamcautionrobotics/robot/Robot.java

```diff
@@ -139,13 +142,20 @@ public void robotPeriodic() {
         } else {
             cargo.intake(CargoMoverSetting.STOP);
         }
-
+        
         if (deployedFunnelRoller != driverRight.getTrigger() && driverRight.getTrigger()) {
             cargo.toggleFunnelRoller();
         }
         deployedFunnelRoller = driverRight.getTrigger();
-
+        
         cargo.deployExitFlap(driverLeft.getTrigger());
+
+        if (jackButtonPressed != manipulator.getButton(Button.X) && manipulator.getButton(Button.X)) {
+            habJack.switchState();
+        }
+        jackButtonPressed = manipulator.getButton(Button.X);
```

Comment:
I really like how this does not need an extra variable to hold the current button state and it only uses one if statement, instead of two in some of my implementations of this logic.

Reply:
Thanks.